src/mcp_agent/utils/resource_utils.py

```
@@ -0,0 +1,222 @@
+    # If it's a simple filename or relative path, convert to file:// URI
+    # Make sure it has three slashes for an absolute path
+    if normalized_path.startswith("/"):
+        return f"file://{normalized_path}"
+    else:
+        return f"file:///{normalized_path}"
+
+
+def extract_title_from_uri(uri: AnyUrl) -> str:
+    """Extract a readable title from a URI."""
+    # Simple attempt to get filename from path
+    uri_str = uri._url
```

Pylint (3.3.7)

[warning] Access to a protected member _url of a client class

(W0212)